compiler/src/dotty/tools/dotc/typer/Checking.scala

```diff
@@ -625,6 +624,24 @@ trait Checking {
       case _ =>
     }
   }
+
+  /** Check that method parameter types do not reference their own parameter
+   *  or later parameters in the same parameter section.
+   */
+  def checkNoForwardDependencies(vparams: List[ValDef])(implicit ctx: Context): Unit = vparams match {
```

Comment:
I'm sure there's a good reason, but it seems that here we are duplicating the checking provided by Config.checkMethodTypes. Why do we need both?